ministryofjustice/analytics-platform-control-panel
# Anpl 862 configure pre commit for dev

pre-commit added to stack

## :memo: Summary
As part of our code cleanup and linting efforts, pre-commit has been added using this ticket.
pre-commit is the main container that holds other tools used to clean and check our code and commits.

This PR ...
- added pre-commit
- added pre-commit configs
- added flake8 & configs
- added isort
- added black
- added commit-msg configuration for message validation

Merging this PR will have the following side-effects:
- no affect in deployment only in local

## :mag: What should the reviewer concentrate on?
- testing commit message format
- test black and isort work by changing something in a python file
- check flake works by seeing if a changed file throws up errors

## :technologist: How should the reviewer test these changes?
- pull changes
- pip install -r requirements.dev.txt
- pre-commit install --hook-type commit-msg
- pre-commit install
- add & commit files as normal
- omit jira ticket number to check jira check is working
- change a python file to have it formatted

## :books: Documentation status
